### PR TITLE
feat(Answer: 1): added solution for the Projection problem

### PR DESCRIPTION
**Pull request by Daes-projet**
## Checklist for challenge submission

- [ ] Start your PR message with Answer:${challenge_number}

### CODE REVIEW

File: apps/angular/projection/src/app/ui/card/card.component.ts [hunk 1]
```diff
@@ -1,34 +1,27 @@
...
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  imports: [NgIf, NgFor, ListItemComponent, NgTemplateOutlet],
 })
-export class CardComponent {
-  @Input() list: any[] | null = null;
-  @Input() type!: CardType;
-  @Input() customClass = '';
+export class CardComponent<T> {
+  @Input({ required: true }) list: T[] | null = null;
```
Comment (tomalaforge) [inline apps/angular/projection/src/app/ui/card/card.component.ts:18]: if it's required, why do you set it to null, prefer initializing list to [] and you can use the empty block of the new @ngfor

File: apps/angular/projection/src/app/component/student-card/student-card.component.ts [hunk 2]
```diff
@@ -1,35 +1,55 @@
...
   ],
-  imports: [CardComponent],
+  imports: [CardComponent, NgIf, AsyncPipe, ListItemComponent],
 })
-export class StudentCardComponent implements OnInit {
-  students: Student[] = [];
-  cardType = CardType.STUDENT;
+export class StudentCardComponent {
+  private readonly _store = inject(StudentStore);
+  protected readonly students$: Observable<Student[]> = this._store.students$;
```
Comment (tomalaforge) [inline apps/angular/projection/src/app/component/student-card/student-card.component.ts:46]: you can now use signals;

**Comment by tomalaforge** [comment]
Good work, I added a few comments.

**Comment by Daes-projet** [comment]
> Good work, I added a few comments.

Thank you for reviewing my code and providing comments on areas for improvement. I've replaced magic strings with directives, and I've experimented with using signals instead of RxJS.

File: apps/angular/projection/src/app/component/city-card/city-card.component.ts [hunk 3]
```diff
@@ -4,30 +4,28 @@ import { CityStore } from '../../data-access/city.store';
...
+    <app-card [list]="cities()" customClass="bg-light-blue">
+      <img src="assets/img/city.webp" width="200px" alt="" />
 
-        <ng-template let-city #cardContent>
-          <app-list-item
-            [name]="city.name"
-            (deleteEmitter)="deleteCity(city.id)">
-          </app-list-item>
-        </ng-template>
+      <ng-template let-city appCardContent>
```
Comment (tomalaforge) [inline apps/angular/projection/src/app/component/city-card/city-card.component.ts:16]: 👍

**Pull request closed without merging**